pkg/fake_gcloud/lib/mem_storage.dart

```diff
@@ -0,0 +1,228 @@
+  @override
+  final String name;
+  final List<int> content;
+  @override
+  final int crc32CChecksum;
+  @override
+  final DateTime updated;
+
+  _File({
+    @required this.bucketName,
+    @required this.name,
+    @required this.content,
+  })  : crc32CChecksum = content.fold<int>(0, (a, b) => a + b) & 0xffffffff,
+        updated = DateTime.now().toUtc();
+
+  @override
+  Uri get downloadLink => Uri(scheme: 'gs', host: bucketName, path: name);
+
+  @override
+  String get etag => crc32CChecksum.toRadixString(16);
+
+  @override
+  ObjectGeneration get generation {
+    throw UnimplementedError();
+  }
+
+  @override
+  int get length => content.length;
+
+  @override
+  List<int> get md5Hash => etag.codeUnits;
+
+  @override
+  ObjectMetadata get metadata {
+    throw UnimplementedError();
+  }
+}
+
+class _Bucket implements Bucket {
+  final _files = <String, _File>{};
+
+  @override
+  final String bucketName;
+
+  _Bucket(this.bucketName);
+
+  @override
+  String absoluteObjectName(String objectName) {
+    if (!objectName.startsWith('/')) {
+      objectName = '/$objectName';
+    }
+    return Uri(scheme: 'gs', host: bucketName, path: objectName).toString();
+  }
+
+  @override
+  StreamSink<List<int>> write(String objectName,
+      {int length,
+      ObjectMetadata metadata,
+      Acl acl,
+      PredefinedAcl predefinedAcl,
+      String contentType}) {
+    if (!objectName.startsWith('/')) {
+      objectName = '/$objectName';
+    }
+    // ignore: close_sinks
+    final controller = StreamController<List<int>>();
+    controller.stream.fold<List<int>>(<int>[], (buffer, data) {
+      buffer.addAll(data);
+      return buffer;
+    }).then((content) {
+      _files[objectName] = _File(
+        bucketName: bucketName,
+        name: objectName,
+        content: content,
+      );
+    });
+    return controller.sink;
+  }
+
+  @override
+  Future<ObjectInfo> writeBytes(String name, List<int> bytes,
+      {ObjectMetadata metadata,
+      Acl acl,
+      PredefinedAcl predefinedAcl,
+      String contentType}) async {
+    if (!name.startsWith('/')) {
+      name = '/$name';
+    }
+    final sink = write(
+      name,
+      metadata: metadata,
+      acl: acl,
+      predefinedAcl: predefinedAcl,
+      contentType: contentType,
+    );
+    sink.add(bytes);
+    await sink.close();
+    return _files[name];
+  }
+
+  @override
+  Future delete(String name) async {
+    if (!name.startsWith('/')) {
+      name = '/$name';
+    }
+    _files.remove(name);
+  }
+
+  @override
+  Future<ObjectInfo> info(String name) async {
+    if (!name.startsWith('/')) {
+      name = '/$name';
+    }
+    final info = _files[name];
+    if (info == null) {
+      throw DetailedApiRequestError(404, '$name does not exists');
+    }
+    return info;
+  }
+
+  @override
+  Stream<List<int>> read(String objectName, {int offset, int length}) async* {
+    if (!objectName.startsWith('/')) {
+      objectName = '/$objectName';
```

Comment:
Does the original API do this... if so leave a comment saying it, otherwise it's probably dangerous to fix the objectName, and better to throw an `ArgumentError`, that way we're forced to fix the code...

Reply:
PTAL: I've changed to required the '/' prefix. I couldn't find if the API returns or requires it, for now it will be fine to pre-validate it.